### ISSUE TITLE
upgrade to play 2.5

### ISSUE THREAD

**Issue by domdorn** [brikis98/ping-play#28]
they've migrated to akka streams.. lib doesn't work anymore :/

**Comment by brikis98** [comment]
Yes, that's a bit unfortunate. I thought their plan had been to maintain backwards compatibility with Iteratee/Enumerator stuff until Play 3.x, but looks like that's not the case. I suspect it's a decent amount of work to do this upgrade and am not sure when I'll be able to get to it. PRs welcome :)

**Comment by Abhinabyte** [comment]
As far I understand, ping-play is based on scala. The javaapi would internally call scala api while rendering streams. And, scala would depend on Play's own Future API for asynchronous programming.<br>
Now Play 2.5  edition is focusing on Java 8 CompletionStage for asynchronous nature(for Java people). And I think Play would entirely move into CompletionStage way in next couple of build. In this case, what to do? Do we need to rebuild entire architecture? Or, some minor patches would do the things?<br>

[](http://stackoverflow.com/questions/38549569/how-to-enable-facebook-style-bigpipe-streaming-in-play-2-5-application-with-java)

**Comment by brikis98** [comment]
@Abhinabyte Play 2.5 introduced [Akka Streams](https://www.playframework.com/documentation/2.5.x/Highlights25#New-streaming-API-based-on-Akka-Streams) as a replacement for Iteratees. That's what ping-play should use as well, both for Java and Scala APIs.
